Refactor e2e

shellwedance opened the pull request:
## Reason for the change
Previously, `hcsctl` installs both rook and cdi module by default, but since `v1.3.2` hcsctl doesn't install cdi module unless specific parameter is given during the install process. Having one test binary, CDI related e2e test cases are still running although CDI module is not installed.

### DONE
- Divide e2e to cdi and ceph
  - Creates two binaries `cdi.test`, `rook.test` after build
- Replace typed client to dynamic client

### TODO
- Add daemonset test
  - Check ceph volume can be attached to every node and read/write data
- Add resize test
  - Check ceph voluem can be resized
- Add snapshot test
  - Check snapshot can be taken

Commit:
Divide e2e to cdi and ceph

Commit:
Apply dynamic client

Commit:
Code refactoring and bug fix

- fix test manifests location
- fix file location in constant
- change e2e package name

SangHyeok-Kim95 commented:
ceph.test is not working. Check the directory path of the ceph e2e manifests.

SangHyeok-Kim95 commented:
lgtm

hbinkim commented on hcsctl/Makefile:
```diff
@@ -6,7 +6,9 @@ check:
 	go build -o build/hcsctl
-	cd e2e && ginkgo build && mv e2e.test ../build/hcsctl.test && cd ..
+	cd e2e/cdi && ginkgo build && mv cdi.test ../../build/cdi.test && cd ..
+	cd e2e/ceph && ginkgo build && mv ceph.test ../../build/ceph.test && cd ..
```
Since we are using `rook` and `cdi` as sample-inventory folder names, and using `[TEST][Rook]` for each e2e test cases, I think having folder name `rook` under `e2e` directory and `rook.test` as a test binary name is more coherent for users.

hbinkim commented on hcsctl/e2e/cdi/cdi_test.go:
```diff
@@ -1,4 +1,4 @@
-package tests
+package testCDI
```
Actually by convention, packages are given lower case, single-word names; there should be no need for underscores or mixedCaps. It's under e2e directory, so I think it's okay to have cdi as a package name. Did you encounter error message for that?

hbinkim commented on hcsctl/e2e/ceph/ceph_test.go:
```diff
@@ -0,0 +1,82 @@
+package testCeph
```
Same as well, by convention, packages are given lower case, single-word names; there should be no need for underscores or mixedCaps. It's under e2e directory, so I think it's okay to have rook as a package name. Did you encounter error message for that?

hbinkim committed:
Update workflow to run e2e with rook.test and cdi.test

- working-directory should be `hcsctl/build` to run rook.test binary with test-manifest folders. Otherwise, while running action pipeline, test-manifest directory can't be found
